packages/xo-web/src/common/xo/index.js

```diff
@@ -1418,7 +1418,13 @@ export const installCertificateOnHost = (id, props) => _call('host.installCertif
 
 export const setControlDomainMemory = (id, memory) => _call('host.setControlDomainMemory', { id, memory })
 
-export const isPubKeyTooShort = id => _call('host.isPubKeyTooShort', { id })
+export const isPubKeyTooShort = (id, powerState, version) => {
+  // this check is only relevant for old hosts, and cannot be done on offline hosts
+  if (semver.satisfies(version, '>=8.3.0') || powerState === 'Halted') {
+    return Promise.resolve(false)
+  }
+  return _call('host.isPubKeyTooShort', { id })
+}
```

Comment:
How about taking an host as parameters (instead of having to pass all parameters one by one)?